app/helpers/selectable_columns_helper.rb

```diff
@@ -34,6 +34,30 @@ def attr_from_callbacks(callbacks, subject)
              .html_safe
   end
 
+  def get_selected_columns
+    column_keys = []
+    Foreman::SelectableColumns::Storage.selected_by(User.current, auto_complete_controller_name).each do |column|
+      column_keys << column['key']
+    end
+    column_keys
+  end
+
+  def defined_columns
+    Foreman::SelectableColumns::Storage.defined_for(auto_complete_controller_name).each do |category|
+      category[:defaultExpanded] = true if category[:id] == 'general'
+      category[:checkProps] = {}
+      category[:children] = category.delete(:columns)
+      category[:children].each { |column| column[:checkProps] = { 'checked': get_selected_columns.include?(column[:id]) } }
```

Comment:
Also, even though it's not so relevant now, I'd suggest to not modify existing saved data, but create a new dataset with needed information instead.